pyomo/contrib/doe/DoE_package.py

```diff
@@ -1805,6 +1876,7 @@ def heatmap(self, title_text, xlabel_text, ylabel_text, font_axes=16, font_tick=
         ba.set_label('log10(cond(FIM))')
         plt.title(title_text + ' - Modified E-optimality')
         plt.show()
+
     
 def sgn(p):
```

Comment:
Is this used anywhere? It is setting j which might be in the global scope. If so, this is considered a dangerous programming practice.

Reply:
This is used to calculate the determinant simultaneously. I moved this into the DesignOfExperiment class so that it is not global used.